editor/src/settings/mod.rs

```diff
@@ -165,9 +166,11 @@ impl SettingsData {
     }
 
     fn save(&mut self) -> Result<(), SettingsError> {
-        let file = File::create(Self::full_path())?;
+        let mut file = File::open(Self::full_path())?;
```

Comment:
This will fail if the settings file does not exist. It should be `File::create` as before.